Vizzuality/heco-invest
# refactoring: Use Location model instead of Mosaic enum

This PR merges Mosaic enums and Location data to one entity. At first, I was trying to keep Mosaic enum and find some way how to keep both entities at sync, but all solutions which I have come with were too fragile and they would break sooner or later. In the end, I have decided to completely replace Mosaic enum with Location model and refactored all code based on this. Unfortunately, there are some breaking changes for API (@barbara-chaves , @clementprdhomme).

**Breaking changes:**
 - there are no more mosaic data as part of `/api/v1/enums`
 - `location_type` filter for `/api/v1/locations` is changed from `region` to `priority_landscape`. To obtain all priority landscapes, please use this endpoint with `filter[location_type]=priority_landscape`
 - `mosaics` parameter when creating or updating project developer is replaced by `priority_landscape_ids` param
 - same is true for project developer API response, `mosaics` fields is replaced everywhere by `priority_landscapes`

There are three new migrations:
 - renaming location type from `region` to `priority_landscape`
 - removing `mosaics` attribute from `ProjectDeveloper` model
 - adding `visible` attribute to `Location`
 
There are also two new rake tasks which needs to be run after this PR is deployed:
 - `rake priority_landscapes:translate` --> adds translation for known priority landscapes
 - `rake priority_landscapes:hide` --> hides priority landscapes that we have decided to ignore for now from API 

## Testing instructions

To manually test this you should:
 - verify Backoffice --> you can change Priority Landscapes for PD and you can filter by priority landscape at project
 - check Rswag --> influenced endpoints are creation of PD and list locations filtered by `priority_landscape` type

## Tracking

https://vizzuality.atlassian.net/browse/LET-583
